src/goto-symex/symex_assign.cpp

```diff
@@ -33,6 +36,29 @@ constexpr bool use_update()
 #endif
 }
 
+/// Determine whether the RHS expression is a string constant initialization
+/// \param rhs The RHS expression
+/// \return True if the expression points to the first character of a string
+///    constant
+static bool is_string_constant_initialization(const exprt &rhs)
+{
+  if(rhs.id() == ID_address_of)
+  {
+    const address_of_exprt &address_of = to_address_of_expr(rhs);
+    if(address_of.object().id() == ID_index)
+    {
+      const index_exprt &index = to_index_expr(address_of.object());
+      if(
+        index.array().id() == ID_string_constant &&
+        index.index() == from_integer(0, c_index_type()))
```

Comment:
Does any other type not work here? Could you instead use `index.index().is_zero()`?